src/protocols/Protocols.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <protocols/bdx/BdxMessages.h>
+#include <protocols/echo/Echo.h>
+#include <protocols/interaction_model/Constants.h>
+#include <protocols/secure_channel/Constants.h>
+#include <protocols/user_directed_commissioning/UserDirectedCommissioning.h>
+
+namespace chip {
+namespace Protocols {
+
+static const char * LookupMessageTypeName(const MessageTypeNameLookup * lookupTable, size_t tableSize, uint8_t msgType)
+{
+    for (auto ptr = lookupTable; ptr != (lookupTable + tableSize); ptr++)
+    {
+        if (ptr->mId == msgType)
+        {
+            return ptr->mName;
+        }
+    }
+
+    return nullptr;
+}
+
+const char * GetProtocolName(Id protocolId)
+{
+    if (protocolId.GetVendorId() != VendorId::Common)
+    {
+        return nullptr;
+    }
+
+    switch (protocolId.GetProtocolId())
+    {
+    case InteractionModel::Id.GetProtocolId():
+        return InteractionModel::kName;
+        break;
+
+    case SecureChannel::Id.GetProtocolId():
+        return SecureChannel::kName;
+        break;
+
+    case BDX::Id.GetProtocolId():
+        return bdx::kName;
+        break;
+
+    case Echo::Id.GetProtocolId():
+        return Echo::kName;
+        break;
+
+    case UserDirectedCommissioning::Id.GetProtocolId():
+        return UserDirectedCommissioning::kName;
+        break;
+
+    default:
+        return nullptr;
+    }
+}
+
+const char * GetMessageTypeName(Id protocolId, uint8_t msgType)
```

Comment:
I believe the functions in this file should be updated to never return nullptr. These are used in `%s` formatting and I am unsure if it is guaranteed that printf of null is acceptable.

Please update them to return `UNNKNOWN` or `UNDEFINED` or similar instead of nullptr.

Reply:
The return value of that is checked against nullptr, so printf is not actually ever being passed a null. Nonetheless, will update it.